apps/dashboard/src/app/team/[team_slug]/[project_slug]/connect/analytics/_components/DailyConnectionsChartCard.tsx

```diff
@@ -0,0 +1,148 @@
+import {
+  type ChartConfig,
+  ChartContainer,
+  ChartTooltip,
+  ChartTooltipContent,
+} from "@/components/ui/chart";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import type { WalletStats } from "@3rdweb-sdk/react/hooks/useApi";
+import { format } from "date-fns";
+import { useMemo, useState } from "react";
+import { Bar, BarChart, CartesianGrid, LabelList, XAxis } from "recharts";
+import { EmptyChartState, LoadingChartState } from "./EmptyChartState";
+
+type ChartToShow = "uniqueWallets" | "totalWallets";
+
+const chartConfig = {
+  uniqueWallets: {
+    label: "Unique Wallets",
+    color: "hsl(var(--chart-1))",
+  },
+  totalWallets: {
+    label: "Total Wallets",
+    color: "hsl(var(--chart-1))",
+  },
+} satisfies ChartConfig;
+
+export function DailyConnectionsChartCard(props: {
+  walletStats: WalletStats;
+  isLoading: boolean;
+}) {
+  const { walletStats } = props;
+
+  const [chartToShow, setChartToShow] = useState<ChartToShow>("uniqueWallets");
+  const chartToShowOptions: ChartToShow[] = ["uniqueWallets", "totalWallets"];
+  const chartLabelToShow: Record<ChartToShow, string> = {
+    uniqueWallets: "Unique Wallets",
+    totalWallets: "Total Wallets",
+  };
+
+  type ChartData = {
+    time: string; // human readable date
+    totalWallets: number;
+    uniqueWallets: number;
+  };
```

Comment:
move out of the component itself (no need to re-instantiate on every render)

Reply:
It's just a type, but sure